src/lualib/ArraySetLength.ts

```diff
@@ -0,0 +1,10 @@
+function __TS__ArraySetLength<T>(this: void, arr: T[], length: number): number {
+    if (length < 0) {
```

Comment:
In JS it fails for any non-integers (32 bit), which also includes `.length = 0.1`, `.length = Infinity` and `.length = NaN`.